### ISSUE TITLE
Hide API docu for private/internal constructs on website

### ISSUE THREAD

**Issue by elfenpiff** [eclipse-iceoryx/iceoryx#745]
## Brief feature description

At the moment the website shows the full API documentation of constructs which are filed under the directory "internal", are private/protected nested classes or methods.

The website should only contain the public user API documentation.

## Detailed information

Adjust the doxygen configuration files so that it is not generated when:
- the class is a protected/private nested class of some class
- the class/function/variable is in the namespace internal
- the methods/variables are protected or private
- the class is stored under the directory internal

**Comment by elBoberido** [comment]
@mossmaurice I strongly suggest to rethink this. The search functionality also lists the private header which makes it hard for users to easily grasp what they can use. Furthermore, try to search for the publisher API. It's a mess.

Oh, and I just noticed that doxygen creates the wrong include paths. Look here https://iceoryx.io/v1.0.1/API-reference/posh/Classes/structiox_1_1popo_1_1PublisherOptions/. It's `#include <publisher_options.hpp>` but should be `include <iceoryx_posh/popo/publisher_options.hpp>`. Do you know if this can be fixed? if not it might be better to nod have it there.

**Comment by elfenpiff** [comment]
@elBoberido I completely agree! And sadly somehow I did not realize that this issue was closed therefore I intervene now and reopen the issue.

**Comment by mossmaurice** [comment]
Still 3:2 for closing this issue :stuck_out_tongue: Just kidding, I [added](https://github.com/eclipse-iceoryx/iceoryx/wiki/2021-09-16-Eclipse-iceoryx-developer-meetup) it to the agenda of next weeks dev meetup.

> Do you know if this can be fixed?
Not atm, please track this point somewhere, else it'll be forgotten. Maybe @ithier can help?

**Comment by elBoberido** [comment]
@mossmaurice I think this is more of a doxygen issue instead of mkdocs issue. I will create an issue for this

Regarding the other topic. @MatthiasKillat once proposed to slightly adjust the folder structure. I don't fully recall his proposal but it made sense when he did it. It think it was something similar to this
```console
|- include
  |- iceoryx_hoofs
    |- cxx
    |- ...
|- src
  |- include
    |- iceoryx_hoofs
      |- internal
        |- cxx
        |- ... 
```

With this, the include paths would remain the same and we can also install the internal headers to the same location but it's now easy to have only the public header in doxygen.
